pkg/project/api/validation/validation.go

```diff
@@ -52,9 +55,40 @@ func validateNoNewLineOrTab(s string) bool {
 func ValidateProjectUpdate(newProject *api.Project, oldProject *api.Project) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&newProject.ObjectMeta, &oldProject.ObjectMeta).Prefix("metadata")...)
-	allErrs = append(allErrs, validateNodeSelector(newProject)...)
-	newProject.Spec.Finalizers = oldProject.Spec.Finalizers
-	newProject.Status = oldProject.Status
+	allErrs = append(allErrs, ValidateProject(newProject)...)
+
+	if !reflect.DeepEqual(newProject.Spec.Finalizers, oldProject.Spec.Finalizers) {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.finalizers", oldProject.Spec.Finalizers, "field is immutable"))
+	}
+	if !reflect.DeepEqual(newProject.Status, oldProject.Status) {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("status", oldProject.Spec.Finalizers, "field is immutable"))
+	}
+
+	// TODO this restriction exists because our authorizer/admission cannot properly express and restrict mutation on the field level.
+	changedAnnotations := util.StringSet{}
+	for name, value := range newProject.Annotations {
+		if name == projectapi.ProjectDisplayName || name == projectapi.ProjectDescription {
+			continue
+		}
+
+		if value != oldProject.Annotations[name] {
+			changedAnnotations.Insert(value)
```

Comment:
append a NewFieldInvalid error per invalid annotation so the field name actually resolves correctly